projects/fw103/src/main.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Pin A6 Reader Function
+void read_pin_a6(const GpioAddress *address, void *context) {
+  uint16_t pin_a6_data = 0;
+  GpioAddress *pin_a6 = context;
+
+  if (adc_read_converted_pin(*pin_a6, &pin_a6_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("Pin A6 Data: %d\n", pin_a6_data);
+  } else {
+    LOG_DEBUG("Failed to read Pin A6 Data");
+  }
+}
+
+// Main
+int main(void) {
+  // Pin Initializations
+  GpioAddress pin_a6 = { .port = GPIO_PORT_A, .pin = 6 };
+
+  GpioSettings a6_settings = {
+    GPIO_DIR_IN,
+    GPIO_STATE_LOW,
+    GPIO_RES_NONE,
+    GPIO_ALTFN_ANALOG,
+  };
+
+  GpioAddress button_pin = { .port = GPIO_PORT_B, .pin = 2 };
```

Comment:
An alternate way of doing these gpio declarations that we use frequently to reduce clutter is the use of preprocessor statements at the top of the file i.e.:
#define ADC_ADDRESS {.port = GPIO_PORT_A, .pin = 6}
This allows us to in main.c simply just do the assignment as
GpioAddress adc_ad = ADC_ADDRESS which will insert the struct we defined above.

The way you've done it is fine, just a thought :)